src/thing_creature.c

```diff
@@ -5153,12 +5153,12 @@ struct Thing *pick_up_creature_of_model_and_gui_job(long crmodel, long job_idx,
     struct Dungeon* dungeon = get_dungeon(plyr_idx);
     if (crmodel < game.conf.crtr_conf.model_count)
     {
-        if ((job_idx == -1) || (dungeon->guijob_all_creatrs_count[crmodel][job_idx & 0x03]))
+        if ((job_idx == -1) || (((job_idx & 0x03) <= 2) && dungeon->guijob_all_creatrs_count[crmodel][job_idx & 0x03]))
```

Comment:
Why is this change here? What does it do?

Reply:
Just fix the warning reported by the new compilers. "job_idx & 0x03" might be 3, which exceeds the array's boundary.